### PR TITLE
feat: allows to override ingress host

### PR DESCRIPTION
**Pull request by yelhouti**

### CODE REVIEW

File: helm/charts/templates/ingress.yaml [hunk 1]
```diff
@@ -1,4 +1,5 @@
-{{- if and (.Values.jxRequirements.ingress.domain) (not .Values.knativeDeploy) }}
+{{- $host := (.Values.ingress.host | default (printf "%s%s%s" .Values.service.name .Values.jxRequirements.ingress.namespaceSubDomain .Values.jxRequirements.ingress.domain)) | quote }}
```
Comment (msvticket) [inline helm/charts/templates/ingress.yaml:1]: $host should be set within the if statement, as the other variables only needed if the ingress is actually created.

File: helm/charts/templates/ingress.yaml [hunk 2]
```diff
@@ -1,4 +1,5 @@
...
+{{- if or (.Values.ingress.host) (and (.Values.jxRequirements.ingress.domain) (not .Values.knativeDeploy)) }}
```
Comment (msvticket) [inline helm/charts/templates/ingress.yaml:2]: The ingress should never be created if knativeDeploy is true

File: helm/charts/templates/ingress.yaml [hunk 3]
```diff
@@ -31,15 +32,20 @@ spec:
...
+    host: {{ $host }}
 {{- end }}
 {{- if .Values.jxRequirements.ingress.tls.enabled }}
   tls:
   - hosts:
-    - {{ .Values.service.name }}{{ .Values.jxRequirements.ingress.namespaceSubDomain }}{{ .Values.jxRequirements.ingress.domain }}
+    - {{ $host }}
+{{- if or (.Values.ingress.tlsWwwPrefix) (eq (index .Values.ingress.annotations "nginx.ingress.kubernetes.io/from-to-www-redirect") "true" ) }}
+    - www.{{ .Values.ingress.host }}
+{{- end }}
```
Comment (msvticket) [inline helm/charts/templates/ingress.yaml:44]: This is an unrelated change that shouldn't be in this PR

**Review by msvticket — CHANGES_REQUESTED** [review]
Thank you for the PR!

Please address my comments! I only checked and added the comments to the "parent" chart, but they are are of course valid for all.

**Comment by yelhouti** [comment]
@msvticket is it possible to generate the child charts from the parent one, or do I need to change all charts manually ?

**Comment by msvticket** [comment]
> is it possible to generate the child charts from the parent one, or do I need to change all charts manually ? 

In each "child" chart directory there is a Kptfile referring to the "parent" chart. So it seems like the intended work flow is to first merge the change to /helm/charts and then use `kpt pkg update` in each child chart to create a new PR.

> I modified all the charts manually could you please review

That works as well 😌

**Review by msvticket — APPROVED** [review]
Now it looks good. Thanks!

**Pull request merged**